lib/modules/apostrophe-pieces-headless/index.js

```diff
@@ -253,13 +253,32 @@ module.exports = {
 
     self.findForRestApi = function(req) {
       var which = 'public';
+      var excludedFields = {};
+
       if (self.apos.permissions.can(req, 'edit-' + self.name)) {
         which = 'manage';
+      } else {
+        self.schema.forEach(function(field) {
+          if (field.privateOnlyForPublic) {
```

Comment:
See my comments above re: calling this property `api`

Reply:
@boutell I did not understand. Did you mean I should change the name `privateOnlyForPublic`?

Reply:
Yes it is super confusing (:

The property should just be called `api` and be set explicitly to `false`, or to the string `'loginRequired'`, depending on whether the API doesn't see it at all or sees it only when there is a user. Otherwise it's visible to the API.